src/prefect/server/models/artifacts.py

```diff
@@ -172,6 +184,44 @@ async def delete_artifact(
     Returns:
         bool: True if the delete was successful, False otherwise
     """
+    current_artifact = await session.get(db.Artifact, artifact_id)
+    if current_artifact is None:
+        return False
+
+    is_latest_artifact_version = (
+        await session.execute(
+            sa.select(db.ArtifactCollection)
+            .where(db.ArtifactCollection.key == current_artifact.key)
+            .where(db.ArtifactCollection.latest_id == artifact_id)
+        )
+    ).scalar_one_or_none()
+
+    if is_latest_artifact_version is not None:
+        previous_artifact = (
```

Comment:
Maybe this should be `next_latest_artifact_version` or similar?

Reply:
That sounds better for sure